crates/bevy_ecs/src/schedule/run_criteria.rs

```diff
@@ -47,13 +47,16 @@ pub enum ShouldRun {
 pub(crate) struct BoxedRunCriteria {
     criteria_system: Option<BoxedSystem<(), ShouldRun>>,
     initialized: bool,
+    archetype_generation: ArchetypeGeneration,
 }
 
 impl Default for BoxedRunCriteria {
     fn default() -> Self {
         Self {
             criteria_system: None,
             initialized: false,
+            // MAX ensures access information will be initialized on first run.
+            archetype_generation: ArchetypeGeneration::new(usize::MAX),
```

Comment:
Note, this conflicts with #2172

Reply:
With that PR, `usize::MAX` is no longer a special case for `ArchetypeGeneration`

Reply:
yup, this PR fixes the issue as it is in current main, I'll update it if #2172 is merged first

Reply:
updated